### PR TITLE
Version with presented config json in INspector DOM  area

### PR DESCRIPTION
**Pull request by sashkivna**

### CODE REVIEW

File: scripts/main/designer/components/inspector/inspector-component/Fields/Text/inspector-text-component.directive.js [hunk 1]
```diff
@@ -2,6 +2,32 @@ angular.module('designer.module')
...
+                                            var description = $scope.$root.definitions.components[i].inspectorConfig[j][key];
+                                        }
+                                    }
+                                }
+                            }*/
+
+                            $scope.finalJSON.push({
+                                _name: $scope.$root.definitions.components[i]._name,
+                                _type: $scope.$root.definitions.components[i]._type,
+                                label: $scope.$root.definitions.components[i].label,
```
Comment (r0ss0neri) [inline scripts/main/designer/components/inspector/inspector-component/Fields/Text/inspector-text-component.directive.js]: `label` and `description` are hard-coded here 

as I mentioned before, try to not relay on prop names

Comment (r0ss0neri) [reply scripts/main/designer/components/inspector/inspector-component/Fields/Text/inspector-text-component.directive.js]: also, finalJSON must be in `designer.directive`

File: scripts/main/view-components/foo/foo.constant.js [hunk 2]
```diff
@@ -9,8 +9,7 @@ angular.module('designer.module')
                     type: 'text'
                 },
                 {
-                    name: 'id',
-                    type: 'number'
+                    description: "blabla"
```
Comment (r0ss0neri) [inline scripts/main/view-components/foo/foo.constant.js]: name: 'description'

    {
       name: 'label',
       type: 'text'
    }
    ...
    {
       name: 'someOtherProperty',
       type: 'someType' // e.g number
    }

**Review by r0ss0neri — CHANGES_REQUESTED** [review]

File: scripts/main/designer/components/inspector/inspector-component/Fields/Text/inspector-text-component.directive.js (hunk 1 above)
Comment (r0ss0neri) [reply scripts/main/designer/components/inspector/inspector-component/Fields/Text/inspector-text-component.directive.js]: particular inspector field must have only logic to write value to the model, he must not aware about some finalJSON

responsibility for this filed to accept type and model and render field bused on the type and than write value to the model

**Review by r0ss0neri — APPROVED** [review]

File: scripts/index.html [hunk 3]
```diff
@@ -12,15 +12,18 @@
 <script src="../node_modules/angular/angular.js"></script>
 <script src="../node_modules/jquery/dist/jquery.js"></script>
 <script src="app-module.js"></script>
+<script src="main/view-components/provider/foo.config.js"></script>
+<script src="main/view-components/provider/registerComponents.service.js"></script>
```
Comment (r0ss0neri) [inline scripts/index.html]: `register-components.service.js`

File: scripts/main/designer/components/inspector/inspector-component/Fields/Number/inspector-number-component.directive.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+angular.module('designer.module')
+    .directive('numberField', function () {
+        return {
+            restrict: 'E',
+            templateUrl: 'main/designer/components/inspector/inspector-component/Fields/Number/inspector-number-component.template.html'
```
Comment (r0ss0neri) [inline scripts/main/designer/components/inspector/inspector-component/Fields/Number/inspector-number-component.directive.js]: don't use camel case for file and folder names

**Review by r0ss0neri — APPROVED** [review]

**Comment by r0ss0neri** [comment]
👍 🥇

File: scripts/index.html (hunk 3 above)
Comment (sashkivna) [reply scripts/index.html]: done

File: scripts/main/designer/components/inspector/inspector-component/Fields/Number/inspector-number-component.directive.js (hunk 4 above)
Comment (sashkivna) [reply scripts/main/designer/components/inspector/inspector-component/Fields/Number/inspector-number-component.directive.js]: done

**Review by r0ss0neri — APPROVED** [review]
👍

**Pull request merged**